File: python/2760.py
from typing import List

class Solution:
    def longestAlternatingSubarray(self, nums: List[int], threshold: int) -> int:
        i = 0
        ans = 0
        while i < len(nums):
            if nums[i] <= threshold and nums[i] % 2 == 0:
                t = 1
                while i + 1 < len(nums):
                    if nums[i+1] <= threshold and (nums[i] + nums[i+1])% 2 == 1:
                        t += 1
                    else:
                        break
                    i += 1
                ans = max(ans, t)
            i += 1
        return ans

File: go/2760.go
package main

func longestAlternatingSubarray(nums []int, threshold int) (ans int) {
	for i := 0; i < len(nums); i++ {
		if nums[i] <= threshold && nums[i]%2 == 0 {
			t := 1
			for ; i+1 < len(nums); i++ {
				if nums[i+1] <= threshold && (nums[i]+nums[i+1])%2 == 1 {
					t++
				} else {
					break
				}
			}
			ans = max(ans, t)
		}
	}
	return
}

func main() {

}
